pkg/build/builder/sti.go

```diff
@@ -33,7 +33,7 @@ func NewSTIBuilder(client DockerClient, dockerSocket string, authCfg docker.Auth
 func (s *STIBuilder) Build() error {
 	tag := s.build.Parameters.Output.DockerImageReference
 	request := &stiapi.Request{
-		BaseImage:    s.build.Parameters.Strategy.STIStrategy.Image,
+		BaseImage:    s.build.Parameters.Strategy.STIStrategy.From.Name,
```

Comment:
This is not going to do what you want.  BaseImage at this point needs to be a valid docker pull spec.  STIStrategy.Image is a valid docker pull spec.  STIStrategy.From.Name is not.  STIStrategy.From.Name points to an ImageRepository object which proxies a valid docker pull spec.

The way I think this whole feature needs to work is that *builds* use the Image field.  BuildConfigs use the From (or optionally Image) field.  When you create a build from a buildconfig that used From, you, at that point in time, resolve the From reference to a docker pull spec, and create a build that has an Image field with that value.

(Keeping in mind that if the build was triggered by an ImageChange trigger, then you need to be using the specific docker pull spec indicated by that ImageChange trigger)